### PR TITLE
feat: New Module `avm/res/app-configuration/configuration-store`

### PR DESCRIPTION
**Pull request by krbar**
## Description

New Module `avm/res/app-configuration/configuration-store`, migrated from CARML.

## Adding a new module

- [x] A proposal has been submitted and approved.
- [ ] I have included "Closes #{module_proposal_issue_number}" in the PR description.
- [ ] I have run `brm validate` locally to verify the module files.
- [x] I have run deployment tests locally to ensure the module is deployable.

## Pipeline references

| Pipeline |
| - |
| [avm.res.app-configuration.configuration-store](https://github.com/krbar/bicep-registry-modules/actions/workflows/avm.res.app-configuration.configuration-store.yml) |

### CODE REVIEW

File: avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+// ============== //
+// Test Execution //
+// ============== //
+
+@batchSize(1)
+module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
+  params: {
```
Comment (ChrisSidebotham) [inline avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep:65]: Are we not including PE  in max test also?

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -8,7 +8,7 @@
 /avm/res/api-management/service/ @Azure/avm-res-apimanagement-service-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/app/container-app/ @Azure/avm-res-app-containerapp-module-owners-bicep @Azure/avm-core-team-technical-bicep
 /avm/res/app/managed-environment/ @Azure/avm-res-app-managedenvironment-module-owners-bicep @Azure/avm-core-team-technical-bicep
-#/avm/res/app-configuration/configuration-store/ @Azure/avm-res-appconfiguration-configurationstore-module-owners-bicep @Azure/avm-core-team-technical-bicep
+/avm/res/app-configuration/configuration-store/ @Azure/avm-res-appconfiguration-configurationstore-module-owners-bicep @Azure/avm-core-team-technical-bicep
```
Comment (AlexanderSehr) [inline .github/CODEOWNERS:11]: Blocking until team is approved

File: avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep (hunk 1 above)
Comment (AlexanderSehr) [reply avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep:65]: We could. Some module may not be able to apply all properties if a PE is deployed, so @krbar would need to check if it works. If it does though, the seperate `pe`  test could be removed

File: avm/res/app-configuration/configuration-store/tests/e2e/pe/main.test.bicep [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+          nestedDependencies.outputs.privateDNSZoneResourceId
+
+        ]
+        subnetResourceId: nestedDependencies.outputs.subnetResourceId
+        tags: {
+          'hidden-title': 'This is visible in the resource name'
+          Environment: 'Non-Prod'
+          Role: 'DeploymentValidation'
+        }
+      }
```
Comment (AlexanderSehr) [inline avm/res/app-configuration/configuration-store/tests/e2e/pe/main.test.bicep:70]: Let's add the 2nd deployment as well to make sure it works :)

File: avm/res/app-configuration/configuration-store/key-value/main.bicep [hunk 4]
```diff
@@ -0,0 +1,40 @@
+metadata name = 'App Configuration Stores Key Values'
+metadata description = 'This module deploys an App Configuration Store Key Value.'
+metadata owner = 'Azure/module-maintainers'
+
+@description('Required. Name of the key.')
+param name string
+
+@description('Required. Name of the value.')
```
Comment (AlexanderSehr) [inline avm/res/app-configuration/configuration-store/key-value/main.bicep:8]:
```suggestion
@description('Required. The value of the key-value.')
```
Isn't it rather this?

File: avm/res/app-configuration/configuration-store/main.bicep [hunk 5]
```diff
@@ -0,0 +1,425 @@
...
+      {
+        categoryGroup: 'AllLogs'
+        enabled: true
+      }
+    ]
+    marketplacePartnerId: diagnosticSetting.?marketplacePartnerResourceId
+    logAnalyticsDestinationType: diagnosticSetting.?logAnalyticsDestinationType
+  }
+  scope: configurationStore
+}]
```
Comment (AlexanderSehr) [inline avm/res/app-configuration/configuration-store/main.bicep:188]:
```suggestion
resource configurationStore_diagnosticSettings 'Microsoft.Insights/diagnosticSettings@2021-05-01-preview' = [for (diagnosticSetting, index) in (diagnosticSettings ?? []): {
  name: diagnosticSetting.?name ?? '${name}-diagnosticSettings'
  properties: {
    storageAccountId: diagnosticSetting.?storageAccountResourceId
    workspaceId: diagnosticSetting.?workspaceResourceId
    eventHubAuthorizationRuleId: diagnosticSetting.?eventHubAuthorizationRuleResourceId
    eventHubName: diagnosticSetting.?eventHubName
    metrics: [for group in (diagnosticSetting.?metricCategories ?? [ { category: 'AllMetrics' } ]): {
      category: group.category
      enabled: group.?enabled ?? true
      timeGrain: null
    }]
    logs: [for group in (diagnosticSetting.?logCategoriesAndGroups ?? [ { categoryGroup: 'allLogs' } ]): {
      categoryGroup: group.?categoryGroup
      category: group.?category
      enabled: group.?enabled ?? true
    }]
    marketplacePartnerId: diagnosticSetting.?marketplacePartnerResourceId
    logAnalyticsDestinationType: diagnosticSetting.?logAnalyticsDestinationType
  }
  scope: configurationStore
}]
```

File: avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep (hunk 1 above)
Comment (krbar) [reply avm/res/app-configuration/configuration-store/tests/e2e/max/main.test.bicep:65]: This is exactly the case, therefore two separate tests.

File: .github/CODEOWNERS (hunk 2 above)
Comment (krbar) [reply .github/CODEOWNERS:11]: This should be fine now.

File: avm/res/app-configuration/configuration-store/tests/e2e/pe/main.test.bicep (hunk 3 above)
Comment (krbar) [reply avm/res/app-configuration/configuration-store/tests/e2e/pe/main.test.bicep:68]: Added a second PE, which didn't went well at first. I had to limit the batchSize to 1.
Commit: https://github.com/Azure/bicep-registry-modules/pull/1072/commits/406b5090ceee1502bf2b752c3abd4e90909c4967

**Review by ChrisSidebotham — APPROVED** [review]

**Pull request merged**